### PR TITLE
Atom-bomb decay, improved vertical and horizontal diffusion

### PR DESCRIPTION
**Pull request by heikoklein**
SNAP is with this PR able to handle bomb-decay of the form proposed by Glasstone-Dolan: d_1 = d_0*t^-1.2 The usefulness of this method is still under testing.

Diffusion is now independent on model timestep by assuming a gaussian distribution and therefore being dependent on sqrt(timestep). This enlarges the diffusion slightly for shorter timesteps, i.e. < 15min for vertical diffusion and < 60min for horizontal diffusion.

The vertical diffusion for timesteps < 15min and within the boundary layer has been improved by using the real boundary-layer height in eta-levels rather than scaling the eta-levels to the boundary-layer height in m. In addition, the diffusion within the boundary layer allows now a 10% entrainment into the free atmosphere. This was documented already, but only implemented in the code for timesteps > 15min/full boundary layer mixing. 

The new diffusion code was tested against the etex case, giving a much improved bias from nmb > 2 to nmb ~ 0.2 (and a minimal improvement against full boundary-layer mixing code (0.17 vs 0.23)).

### CODE REVIEW

File: src/common/rwalk.f90 [hunk 1]
```diff
@@ -99,18 +107,18 @@ subroutine rwalk(blfullmix,part,pextra)
...
-    if (blfullmix) then
-      part%z = 1.0 - (1.0 - part%tbl)*1.1*(rnd(3)+0.5)
-    else ! not full mixing
-      hfactor = part%hbl/hmax
-      rv = lmax*hfactor*tfactor
-      rvmax = 1.0 - part%tbl
-
-      rv = min(rv, rvmax)
+    if (blfullmix .or. (tsqrtfactor_v .gt. 1.0)) then ! full mixing
+      part%z = 1.0 - (1.0 - part%tbl)*(1.+entrainment)*(rnd(3)+0.5)
```
Comment (magnusuMET) [inline src/common/rwalk.f90:111]: Move factor `(1.0 - part%tbl)*(1.+entrainment)` outside if and give a name to it (top_of_entrainment_zone?). Do you need to check if negative as on l.120?

**Review by magnusuMET — APPROVED** [review]
Only have some minor nits, otherwise looks great!

File: src/common/decay.f90 [hunk 2]
```diff
@@ -50,30 +50,53 @@ subroutine decayDeps(tstep)
...
+  USE iso_fortran_env, only: real64
 
   real, intent(in) :: tstep
 
   integer :: m, mm
+  real :: bomb_decay_rate, current_state, next_state
   logical, save :: prepare = .TRUE.
+  logical, save :: has_bomb_decay = .FALSE.
+  !> totalstep in seconds run in decay
+  !> start at 1h to satisfy t[h]^-1.2 initial state
```
[the remark this answers is not in the document]
Comment (heikoklein) [reply src/common/decay.f90]: Done

File: src/common/rwalk.f90 (hunk 1 above)
Comment (heikoklein) [reply src/common/rwalk.f90]: Done

File: src/common/rwalk.f90 [hunk 3]
```diff
@@ -39,11 +44,14 @@ subroutine rwalk_init(timestep)
 !> time step in seconds (trajectory calculations)
   real, intent(in) :: timestep
 
-  tfactor = timestep/tmix
+  tfactor_v = timestep/tmix_v
+  tsqrtfactor_v=1./sqrt(1./tfactor_v)
```
[the remark this answers is not in the document]
Comment (heikoklein) [reply src/common/rwalk.f90]: Yes

File: src/common/rwalk.f90 [hunk 4]
```diff
@@ -39,11 +44,14 @@ subroutine rwalk_init(timestep)
...
+  tfactor_h = timestep/tmix_h
+  tsqrtfactor_h=1./sqrt(1./tfactor_h)
```
[the remark this answers is not in the document]
Comment (heikoklein) [reply src/common/rwalk.f90]: Yes

**Pull request merged**